MoreLinq.Test/AggregateRightTest.cs

```diff
@@ -0,0 +1,124 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    public class AggregateRightTest
+    {
+        // Overload 1 Test
+
+        [Test]
+        public void AggregateRight1WithNullSequence()
+        {
+            Assert.ThrowsArgumentNullException("source", 
+                () => MoreEnumerable.AggregateRight<int>(null, (a, b) => a + b));
+        }
+
+        [Test]
+        public void AggregateRight1WithNullFunc()
+        {
+            Assert.ThrowsArgumentNullException("func", 
+                () => Enumerable.Range(1, 5).AggregateRight(null));
+        }
+
+        [Test]
+        public void AggregateRight1WithEmptySequence()
+        {
+            Assert.Throws<InvalidOperationException>(
+                () => new int[] { }.AggregateRight((a, b) => a + b));
+        }
+
+        [Test]
+        public void AggregateRight1SimpleTest()
+        {
+            var numbersAsString = Enumerable.Range(1, 5).Select(x => x.ToString());
+
+            Assert.That(numbersAsString.AggregateRight((a, b) => string.Format("({0}+{1})", a, b)), Is.EqualTo("(1+(2+(3+(4+5))))"));
```

Comment:
This is hard to read with too much going on on a single line. What's more, `AggregateRight` that's the real subject under test here belongs with the `Range` and `Select` chain. I would recommend rewriting as:

```c#
var result = Enumerable.Range(1, 5)
                       .Select(x => x.ToString())
                       .AggregateRight((a, b) => $"({a}+{b})"));

Assert.That(result, Is.EqualTo("(1+(2+(3+(4+5))))"));
```

For very trivial cases, I don't follow the strictly follow the [act-arrange-assert](http://wiki.c2.com/?ArrangeActAssert) pattern, but here it makes sense for readability to at least split arrange-act and assert.